app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,49 @@
+class CategoriesController < ApplicationController
+  def index
+    @categories = Category.all
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      redirect_to category_path(@category)
+    else
+      redirect_to new_category_path
+    end
+  end
+
+  def show
+    @category = Category.find(params[:id])
+    @format   = @category.arrange
```

Comment:
it's not clear what this is doing simply from reading this code. be explicit as possible!